docs/DeveloperGuide.md

```diff
@@ -4,26 +4,162 @@
+4. Afterwards, the User Interface class will call addGoods() method from the Commands Class
+5. This method will then call the constructor of the Good class, creating a new instance of the Good Object
+6. The addGood() method will then add the Good to the Collections storing the Good Objects currently in the WareHouse
+
+For more examples of how a user can use a command, refer to the [UserGuide](/UserGuide.md)
+
+### Remove Goods Method
+#### Description
+Remove Goods belongs as part of the Commands Class. It is used to remove a certain amount of goods from the inventory.
+#### Operation
+![removeGood diagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T09-4/tp/master/docs/diagrams/removeGood.puml)
+The above sequence diagram shows the operation of how the add goods method will be called.
+1. The User input will be read by the User Interface Class
+2. The User Interface Class will then match the command keyword `remove`
+3. The Regex Class will then be called to match the rest of the user's input to find the values required to remove goods from the inventory.
+4. Afterwards, the User Interface class will call removeGood() method from the Commands Class
+5. This method will then reduce the quantity of a type of goods if the quantity input is not larger than the existing quantity. If the quantity input is the same as the existing quantity, the goods object will be removed from the inventory.
+6. The UI will show a message of format
+    > `quantity` `name` have been removed
+to show that the operation is successful.
+
+For more examples of how a user can use a command, refer to the [UserGuide](/UserGuide.md)
+
+### List Goods or List Orders Method
+#### Description
+The list goods or list orders method belongs to the Warehouse Class. It is used to display the list of existing goods 
+or existing orders in the warehouse to the User.
+
+![List sequence Diagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T09-4/tp/master/docs/diagrams/List.puml)
+
+The above sequence diagram shows the operation of how the list goods method will be called.
+
+1. The User input will be read by the User Interface Class
+2. The User Interface Class will the match the command keyword `list`
+3. Together with the previous step, the User Interface Class will check the flag followed by `list`
+   3.1 If the flag is `o/`, the User Interface Class will call the listOrders() method in the Warehouse class
+      3.1.1 Lastly, the list of orders is printed to the command line.
+   3.2 Else, if the flag is `g/`, the User Interface Class will call the listGoods() method in the Warehouse class
+      3.2.1 Following that, the Warehouse class will call the getGoods() method on the Good class
+      3.2.2 Lastly, the list of goods is printed to the command line.
```

Comment:
Consider adding \<br> lines to properly format these numbered lists, as they currently appear as one big paragraph.
![image](https://user-images.githubusercontent.com/60597985/160962746-578f4c7f-86f0-47d6-95f6-5989f1b0cbd7.png)